api/internal/handler/service/service.go

```diff
@@ -218,8 +220,34 @@ type BatchDelete struct {
 
 func (h *Handler) BatchDelete(c droplet.Context) (interface{}, error) {
 	input := c.Input().(*BatchDelete)
+	ids := strings.Split(input.IDs, ",")
+	mp := make(map[string]struct{})
+	for _, id := range ids {
+		mp[id] = struct{}{}
+	}
+
+	ret, err := h.routeStore.List(c.Context(), store.ListInput{
+		Predicate: func(obj interface{}) bool {
+			route := obj.(*entity.Route)
+			if _, exist := mp[utils.InterfaceToString(route.ServiceID)]; exist {
+				return true
+			}
+
+			return false
+		},
+		PageSize:   0,
+		PageNumber: 0,
```

Comment:
why both `pagesize` and `pagenumber` are 0 ?

Reply:
We need to list all routes.

Reply:
`0` means all?

Reply:
In restful APIs, when we omit `size` or `number`, it means to fetch all resources. If we pass `0`, there should return an empty array IMO.

cc @imjoey @tokers to confirm

Reply:
`offset = abs(page - 1) * pageSize`

Reply:
You can see here:

https://github.com/apache/apisix-dashboard/blob/38cf0ba32264d5a7bb4f22bd40ce037e2f66a939/api/internal/core/store/store.go#L210-L225